### PR TITLE
python3Packages.curl-cffi: init at 0.6.3

### PR DESCRIPTION
**Pull request by chuangzhu**
## Description of changes

Python binding for curl-impersonate via cffi. A http client that can impersonate browser tls/ja3/http2 fingerprints.

https://curl-cffi.readthedocs.io/

Dependency of #309402, https://github.com/NixOS/nixpkgs/pull/309345#pullrequestreview-2039788592

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/development/python-modules/curl-cffi/default.nix [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+  ];
+
+  format = "pyproject";
+  build-system = [
+    setuptools
+  ];
+
+  nativeBuildInputs = [
+    cffi
+  ];
```
Comment (lilyinstarlight) [inline pkgs/development/python-modules/curl-cffi/default.nix:35]: Does cffi in `nativeBuildInputs` even do anything here?

**Review by lilyinstarlight — COMMENTED** [review]
The curl-impersonate diff looks good to me, thank you!

I left one question but otherwise the package looks good

**Comment by lilyinstarlight** [comment]
It looks like ofborg didn't attempt to build curl-impersonate because of a misspelling in the commit message. Would you be willing to rebase and fix that?

Alternatively with your permission, I'd be happy to push to your branch myself and fix it, since it's mostly a cosmetic change and I have the ability to do that

File: pkgs/development/python-modules/curl-cffi/default.nix (hunk 1 above)
Comment (chuangzhu) [reply pkgs/development/python-modules/curl-cffi/default.nix:35]: Cffi is used in build script so I guess it's needed for cross-compilation, though I haven't tested whether it works in a cross build yet

Comment (lilyinstarlight) [reply pkgs/development/python-modules/curl-cffi/default.nix:35]: Ah yep! I missed that. It is indeed needed. I've got some follow-up changes I'm cooking up a follow-up PR to fix cross-compilation for the package and fix darwin build as well

**Review by lilyinstarlight — APPROVED** [review]

**Comment by lilyinstarlight** [comment]
@ofborg build python3Packages.curl-cffi

File: pkgs/development/python-modules/curl-cffi/default.nix (hunk 1 above)
Comment (lilyinstarlight) [reply pkgs/development/python-modules/curl-cffi/default.nix:35]: Follow-up PR, if you were curious: #310386

**Pull request merged**